localstack/services/events/packages.py

```diff
@@ -0,0 +1,86 @@
+from localstack.constants import MAVEN_REPO_URL
+from localstack.packages import DownloadInstaller, Package, PackageInstaller
+from localstack.packages.api import InstallTarget, MultiPackageInstaller
+
+# https://central.sonatype.com/artifact/software.amazon.event.ruler/event-ruler
+EVENT_RULER_VERSION = "1.7.3"
+# The dependent jackson.version is defined in the Maven POM File of event-ruler
+JACKSON_VERSION = "2.16.2"
+
+
+class MavenPackageInstaller(DownloadInstaller):
+    """LocalStack package installer for Maven dependencies.
+    Follows the Maven naming conventions: https://maven.apache.org/guides/mini/guide-naming-conventions.html
+    """
+
+    # Example: software.amazon.event.ruler
+    group_id: str
+    # Example: event-ruler
+    artifact_id: str
+    # Example version: 1.7.3
+
+    # Optional installation directory to overwrite the default
+    install_dir: str | None
+
+    def __init__(self, package_url: str, install_dir: str | None = None):
+        """ "The packageURL is easy copy/pastable from the Maven central repository
+        Example package_url: pkg:maven/software.amazon.event.ruler/event-ruler@1.7.3
+        """
+        parts = package_url.split("/")
+        self.group_id = parts[1]
+        sub_parts = parts[2].split("@")
+        self.artifact_id = sub_parts[0]
+        version = sub_parts[1]
+        super().__init__(self.artifact_id, version)
+        self.install_dir = install_dir
+
+    def _get_download_url(self) -> str:
+        group_id_path = self.group_id.replace(".", "/")
+        return f"{MAVEN_REPO_URL}/{group_id_path}/{self.artifact_id}/{self.version}/{self.artifact_id}-{self.version}.jar"
+
+    def _get_install_dir(self, target: InstallTarget) -> str:
+        """Allow to overwrite the default installation directory.
+        This enables bundling transitive dependencies into the same directory.
+        """
+        return self.install_dir or super()._get_install_dir(target)
+
+
+class EventRulerPackage(Package):
+    def __init__(self):
+        super().__init__("EventRulerLibs", EVENT_RULER_VERSION)
+
+    def get_versions(self) -> list[str]:
+        return [EVENT_RULER_VERSION]
+
+    def _get_installer(self, version: str) -> PackageInstaller:
+        return EventRulerPackageInstaller()
+
+
+class EventRulerPackageInstaller(MultiPackageInstaller):
+    def __init__(self):
+        event_ruler = MavenPackageInstaller(
+            f"pkg:maven/software.amazon.event.ruler/event-ruler@{EVENT_RULER_VERSION}"
+        )
+        event_ruler_dir = event_ruler._get_install_dir(InstallTarget.VAR_LIBS)
```

Comment:
@alexrashed How can we solve this alternatively?

The EventRuler package installer aims to bundle all dependent jars for a given event ruler version like this rather than having each jar in its own directory, which would require copying them afterward:
<img width="726" alt="Screenshot 2024-04-12 at 11 32 24" src="https://github.com/localstack/localstack/assets/8161292/27d9d9ef-a78a-4ab7-ae70-37161942c5c5">

Reply:
Hmm, I feel like the usage of the `MultiPackageInstaller` is counter productive. It breaks with the definition of a package, where a package is an independently used piece of software with a name and a version.
I think this is the typical use case when you download Jar files from Maven: You could implement the MavenPackageInstaller in a way that it accepts a name and a version, but also a list of group/artifact IDs with their versions which should be downloaded.

This would mean that the installer has a name and a version (event ruler version X), which consists of multiple maven artifacts to be downloaded.

Does that make sense?

Reply:
Yeah, I see. So basically like this:

```py
        return MavenPackageInstaller(
            f"pkg:maven/software.amazon.event.ruler/event-ruler@{EVENT_RULER_VERSION}",
            f"pkg:maven/com.fasterxml.jackson.core/jackson-annotations@{JACKSON_VERSION}",
            f"pkg:maven/com.fasterxml.jackson.core/jackson-core@{JACKSON_VERSION}",
            f"pkg:maven/com.fasterxml.jackson.core/jackson-databind@{JACKSON_VERSION}",
        )
```

I tried to re-use the modular constructs such as the [DownloadInstaller](https://github.com/localstack/localstack/blob/7c7cd158212dfabc37c9e1a39d0a03e60a124e9a/localstack/packages/core.py#L300) for better consistency but that does not seem possible using this approach.